rpg.py

```diff
@@ -0,0 +1,179 @@
+                    return players
+                else:
+                    return players
+    
+class Enemy(Character):
+    # another subclass of character
+    """this is a subclass of characters specific to team 2"""
+    def __init__(self, name: str, hp: int, attack: int, speed: int):
+        super().__init__(name, hp, attack, speed, team=2)
+
+    def act(self, players):
+        all_enemy_locations = self.get_all_enemies(players)
+        if all_enemy_locations:
+            targeted_index = random.choice(all_enemy_locations)
+            targeted_player = players[targeted_index]
+            damaged_player = self.deal_damage(targeted_player)
+            players[targeted_index] = damaged_player
+            return players
+        else:
+            return players
+
+
+@dataclass(order=True)
+class Move:
+    """Handles the priority Queue"""
+    priority: int
+    player: Character = field(compare=False)
+    
+
+class Battle:
+    """list of characters, and order of moves"""
+
+    def __init__(self, players: List[Character]):
+        self.players = players
+        self.battle_queue = PriorityQueue()
+
+    def add_into_queue(self, player: Character, game_time: int) -> None:
+        """adds a player back based on game time
+        faster players go first"""
+        move = Move(priority=game_time + 1/player.speed, player=player)
+        self.battle_queue.put(move)
+
+    def get_from_queue(self) -> Tuple[Character, int]:
+        """removes the player from the queue to add back after cooldown"""
+        move = self.battle_queue.get()
+        return move.player, move.priority
+
+    def is_over(self) -> bool:
+        """true if the battle is over"""
+        return self.battle_queue.empty()
+
+    def run(self):
+        """Makes the battle loop while it's not over"""
+        print('Battle Begin!')
+        print(f'Combatants: {[player.name for player in self.players]}')
+        for player in self.players:
+            self.add_into_queue(player=player, game_time=0)
+
+        while not self.is_over():
+            acting_player, current_game_time = self.get_from_queue()
+            if acting_player.is_alive():
+                updated_players = acting_player.act(self.players)
+                self.players = updated_players
+                for player in self.players:
+                    if player.is_alive():
+                        print(player.name + ':', player.hp)
+            if acting_player.is_alive and acting_player.get_all_enemies(self.players):     
+                self.add_into_queue(acting_player, current_game_time)
+            else:
+                print(f'{acting_player.name} is dead!')
+
+        print('The battle is over')
+        print('The following players survived:')
+        print(f'{[player.name for player in self.players if player.is_alive()]}')
+        time.sleep(3)
+
+
+if __name__ == '__main__':
+    player1 = Ally(name, 5, 3, 2)
```

Comment:
I mentioned this at the top but I'll do it again. Handle `name` right here. Not at the top of the file.